firefox-ios/Client/Frontend/NativeErrorPage/NativeErrorPageViewController.swift

```diff
@@ -0,0 +1,232 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Common
+import ComponentLibrary
+import Redux
+import Shared
+
+final class NativeErrorPageViewController: UIViewController,
+                                           Themeable,
+                                           ContentContainable {
+    private let model: ErrorPageModel
+    private let windowUUID: WindowUUID
+
+    // MARK: Themable Variables
+    var themeManager: Common.ThemeManager
+    var themeObserver: NSObjectProtocol?
+    var notificationCenter: Common.NotificationProtocol
+    var currentWindowUUID: UUID? {
+        windowUUID
+    }
+
+    private var overlayManager: OverlayModeManager
+    var contentType: ContentType = .nativeErrorPage
+
+    // MARK: UI Elements
+    private struct UX {
+        static let logoSizeWidth: CGFloat = 221
+        static let mainStackSpacing: CGFloat = 24
+        static let textStackSpacing: CGFloat = 16
+        static let portraitPadding = NSDirectionalEdgeInsets(
+            top: 76,
+            leading: 16,
+            bottom: -16,
+            trailing: -16
+        )
+        static let landscapePadding = NSDirectionalEdgeInsets(
+            top: 60,
+            leading: 64,
+            bottom: -16,
+            trailing: -64
+        )
+    }
+
+    private lazy var scrollView: UIScrollView = .build()
+
+    private lazy var scrollContainer: UIStackView = .build { stackView in
+        stackView.spacing = UX.mainStackSpacing
+    }
+
+    private lazy var textStack: UIStackView = .build { stackView in
+        stackView.axis = .vertical
+        stackView.spacing = UX.textStackSpacing
+    }
+
+    private lazy var commonContainer: UIStackView = .build { stackView in
+        stackView.axis = .vertical
+        stackView.spacing = UX.mainStackSpacing
+        stackView.distribution = .fill
+    }
+
+    private lazy var logoImage: UIImageView = .build { imageView in
+        imageView.image = UIImage(
+            named: ImageIdentifiers.NativeErrorPage.noInternetConnection
+        )
+        imageView.contentMode = .scaleAspectFit
+    }
+
+    private lazy var titleLabel: UILabel = .build { label in
+        label.adjustsFontForContentSizeCategory = true
+        label.font = FXFontStyles.Bold.title2.scaledFont()
+        label.numberOfLines = 0
+        label.text = .NativeErrorPage.NoInternetConnection.TitleLabel
```

Comment:
do we want to inject these strings from the model instead?

Reply:
Yes, I will add TODO comment here.